connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -79,7 +80,12 @@ public void schedule(final ConnectorTaskId id, final WorkerSourceTask workerTask
         ScheduledFuture<?> commitFuture = commitExecutorService.scheduleWithFixedDelay(new Runnable() {
             @Override
             public void run() {
-                commit(workerTask);
+                try {
+                    LoggingContext.forOffsets(id);
```

Comment:
I see from previous comments that there used to be `LoggingContext` instances. Given current structure of code, I can see how getting rid of instances and uses `static` methods makes sense, but should we consider still using instances and changing `clear` to `close()` so we can just use try-with-resources? That seems less error-prone than this setup where it'd be easy to forget to put in a `clear` call.

Reply:
Agreed. I think my previous simplification went too far, and AutoCloseable does indeed make it much less error prone. Another benefit of using a try-with-resources and AutoCloseable is that the instance can capture and restore the complete MDC parameter state, and this makes nesting logging contexts that much easier and less error-prone.